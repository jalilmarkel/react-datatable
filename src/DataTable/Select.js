import React from 'react';
import styled, { withTheme } from 'styled-components';
import DropDownIcon from '../icons/Dropdown';

const SelectControl = styled.select`
  cursor: pointer;
  height: 24px;
  min-width: 24px;
  user-select: none;
  padding-left: 8px;
  padding-right: 16px;
  box-sizing: content-box;
  font-size: ${props => props.theme.pagination.fontSize};
  color: ${props => props.theme.pagination.fontColor};
  outline: none;
  border: none;
  background-color: transparent;
  appearance: none;

  .ddown-color {
    fill: red;
  }

  &::-ms-expand {
    display: none;
  }

  &:disabled::-ms-expand {
    background: #f60;
  }
`;

const SelectWrapper = styled.div`
  /* width: 100%; */
  position: relative;
  flex-shrink: 0;
  margin-left: 8px;
  margin-right: 32px;
  font-size: ${props => props.theme.pagination.fontSize};
  color: ${props => props.theme.pagination.fontColor};

  svg {
    top: 1;
    right: 0;
    color: rgba(0, 0, 0, 0.54);
    position: absolute;
    pointer-events: none;
    fill: currentColor;
    width: 1em;
    height: 1em;
    display: inline-block;
    font-size: 24px;
    transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
    user-select: none;
    flex-shrink: 0;
  }
`;

const Select = props => (
  <SelectWrapper>
    <SelectControl {...props} />
    <DropDownIcon />
  </SelectWrapper>
);

export default withTheme(Select);
