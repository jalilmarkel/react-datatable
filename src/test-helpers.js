import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

