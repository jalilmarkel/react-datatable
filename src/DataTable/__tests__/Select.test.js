import 'jest-styled-components';
import React from 'react';
import { cleanup, fireEvent } from 'react-testing-library';
import { renderWithTheme } from '../../test-helpers';
import Select from '../Select';

afterEach(cleanup);

test('should render correctly with default props', () => {
  const { container } = renderWithTheme(
    <Select onChange={jest.fn()} defaultValue={1}>
      <option value={1}>
        {1}
      </option>
      ))
    </Select>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('should call onChange if a change is triggered', () => {
  const onChangeMock = jest.fn();
  const { container } = renderWithTheme(
    <Select onChange={onChangeMock} defaultValue={1}>
      <option value={1}>
        1
      </option>

      <option value={2}>
        2
      </option>
      ))
    </Select>,
  );

  fireEvent.change(container.querySelector('select'), { target: { value: 2 } });

  expect(onChangeMock).toBeCalled();
});
