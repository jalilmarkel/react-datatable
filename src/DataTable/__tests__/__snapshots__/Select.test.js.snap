// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default props 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  min-width: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 8px;
  padding-right: 16px;
  box-sizing: content-box;
  font-size: 13px;
  color: rgba(0,0,0,.54);
  outline: none;
  border: none;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1 .ddown-color {
  fill: red;
}

.c1::-ms-expand {
  display: none;
}

.c1:disabled::-ms-expand {
  background: #f60;
}

.c0 {
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 8px;
  margin-right: 32px;
  font-size: 13px;
  color: rgba(0,0,0,.54);
}

.c0 svg {
  top: 1;
  right: 0;
  color: rgba(0,0,0,0.54);
  position: absolute;
  pointer-events: none;
  fill: currentColor;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 24px;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<div
  class="c0"
>
  <select
    class="c1"
  >
    <option
      selected=""
      value="1"
    >
      1
    </option>
    ))
  </select>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</div>
`;
